tck/src/main/java/org/eclipse/microprofile/health/tck/MultipleLivenessFailedTest.java

```diff
@@ -0,0 +1,128 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.eclipse.microprofile.health.tck;
+
+import java.io.StringReader;
+
+import javax.json.*;
+
+import org.eclipse.microprofile.health.tck.deployment.FailedLiveness;
+import org.eclipse.microprofile.health.tck.deployment.FailedReadiness;
+import org.eclipse.microprofile.health.tck.deployment.SuccessfulLiveness;
+import org.eclipse.microprofile.health.tck.deployment.SuccessfulReadiness;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.shrinkwrap.api.Archive;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import static org.eclipse.microprofile.health.tck.DeploymentUtils.createWarFileWithClasses;
+import static org.eclipse.microprofile.health.tck.JsonUtils.asJsonObject;
+
+/**
+ * @author Heiko Braun
+ * @author Antoine Sabot-Durand
+ */
+public class MultipleLivenessFailedTest extends SimpleHttp {
+
+    @Deployment(testable = false)
+    public static Archive getDeployment() throws Exception {
+        return createWarFileWithClasses(FailedLiveness.class,
+                                        SuccessfulLiveness.class,
+                                        SuccessfulReadiness.class);
+    }
+
+    /**
+     * Verifies the liveness health integration with CDI at the scope of a server runtime
+     */
+    @Test
+    @RunAsClient
+    public void testFailureResponsePayload() throws Exception {
+        Response response = getUrlLiveContents();
+
+        // status code
+        Assert.assertEquals(response.getStatus(),503);
+
+        JsonReader jsonReader = Json.createReader(new StringReader(response.getBody().get()));
+        JsonObject json = jsonReader.readObject();
+        System.out.println(json);
+
+        // response size
+        JsonArray checks = json.getJsonArray("checks");
+        Assert.assertEquals(checks.size(), 2, "Expected two check responses");
+
+
+        for (JsonValue check : checks) {
+            String id = asJsonObject(check).getString("name");
+            switch (id) {
+                case "successful-check":
+                    verifySuccessPayload(check);
+                    break;
+                case "failed-check":
+                    verifyFailurePayload(check);
+                    break;
+                default:
+                    throw new IllegalArgumentException("Unexpected response payload structure");
+            }
+        }
+
+        // overall outcome
+        Assert.assertEquals(
+                json.getString("status"),
+                "DOWN",
+                "Expected overall status to be unsuccessful"
+        );
+    }
+
+    private void verifyFailurePayload(JsonValue check) {
+        // single procedure response
+        Assert.assertEquals(
```

Comment:
you're right but as this code is here since since 1.0.
As said in my first comment, this PR shouldn't be used to fix existing test but to check that new tests cover correctly addition of liveness and readiness.
This matter should be treated in a separate ticket for RC1 IMO.